activerecord/lib/arel/nodes/use_index.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+module Arel # :nodoc: all
+  module Nodes
+    class UseIndex < Unary
+      def indexes
+        expr[0]
+      end
+
+      FOR_OPTIONS = { join: "JOIN", order: "ORDER BY", group: "GROUP BY" }.freeze
+      private_constant :FOR_OPTIONS
+      def for
+        FOR_OPTIONS[expr[1].to_sym] if expr[1].respond_to?(:to_sym)
```

Comment:
In case `expr[1]` doesn't respond to `to_sym` we are returning `nil` here. Is it intentional?

Reply:
Is intentional: when an invalid option/type is given `.for` returns `nil` hence the `FOR` part won't be added in the output by the visitor.